### PR TITLE
Smb hashmap/v5

### PR DESCRIPTION
**Pull request by victorjulien**
Adds use of the `lru` crate to turn `guid2name_map` into a `LruCache`. In this test code it's hard coded to 1024 entries max.

Continuation of #11721.

PR for initial LRU usage review and qa feedback.

Todos:
- [x]  config for guid2name limit (1024 hardcoded now)
- [x] undo timestamp storage in guid2name
- [x] git cleanup
- [ ] review comments
- [ ] docs

### CODE REVIEW

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on build_asan.
  

Pipeline 22678

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPR1_stats_chk | | |
| .uptime | 649 | 628 | 96.76% | 

Pipeline 22679

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 22681

File: rust/src/smb/smb1.rs [hunk 1]
```diff
@@ -732,7 +734,7 @@ fn smb1_response_record_one(state: &mut SMBState, r: &SmbRecord, command: u8, an
                             fid.extend_from_slice(&u32_as_bytes(r.ssn_id));
                             SCLogDebug!("SMB1_COMMAND_NT_CREATE_ANDX fid {:?}", fid);
                             SCLogDebug!("fid {:?} name {:?}", fid, p);
-                            state.guid2name_map.insert(fid, p);
+                            _ = state.guid2name_map.put(fid, p);
```
Comment (catenacyber) [inline rust/src/smb/smb1.rs:737]: Should we handle somehow the return value ?

Comment (victorjulien) [reply rust/src/smb/smb1.rs:737]: Not sure what to do with it. IIRC it returns the evicted value.

Comment (jasonish) [inline rust/src/smb/smb1.rs:737]: Perhaps for consistency with `let _name`, use `let _evicted`?

**Comment by jasonish** [comment]
Just for observation sake, this brings in some new crates:
- ahash: this one is new to me
- hashbrown: common, well supported
- once_cell: I believe this to be common and well supported to

**Comment by jasonish** [comment]
Still a draft, but us of the lru crate looks good, and in general this crate is in very wide use.

**Comment by victorjulien** [comment]
continued testing in #12028

**Pull request closed without merging**